### PR TITLE
feat(BetterChat): Translate!!!

### PR DESCRIPTION
**Pull request by DataM0del**
No need for https://github.com/CCBlueX/ScriptAPI/blob/master/examples/translator.js anymore!

### CODE REVIEW

**Comment by DataM0del** [comment]
waiting to be detekted

**Comment by DataM0del** [comment]
already detekted, we're going on a good path :clap:

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/ResponseData.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package net.ccbluex.liquidbounce.features.module.modules.misc.betterchat.translate.providers.deepl
+
+interface ResponseData {
+    val translations: List<Translations>
+}
+interface Translations {
+    val detected_source_language: String
```
[the remark this answers is not in the document]
Comment (DataM0del) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/ResponseData.kt]: I need to figure out how to get gson to turn `detectedSourceLanguage` into `detected_source_language` when parsing like how you can use `#[serde(rename = "detected_source_language")]` with Serde in Rust

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/Translate.kt [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import net.ccbluex.liquidbounce.features.module.modules.misc.betterchat.ModuleBetterChat
+
+/**
+ * BetterChat: Translate addition
+ *
+ * Uses Google Translate or DeepL to translate messages
+ * Inspired by https://github.com/Vendicated/Vencord/tree/main/src/plugins/translate/
+ */
+object Translate : ToggleableConfigurable(ModuleBetterChat, "Translate", true) {
+}
```
[the remark this answers is not in the document]
Comment (DataM0del) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/Translate.kt]: No implementation yet

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/DeepL.kt [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        connection.doOutput = true
+        connection.doInput = true
+        val outputStream = connection.outputStream
+        outputStream.write(text.toByteArray())
+        outputStream.close()
+        connection.connect()
+        val inputStream = connection.inputStream
+        val response = inputStream.bufferedReader().use { it.readText() }
+        val gson = GsonBuilder().create()
+        val responseObj = gson.fromJson(response, ResponseData::class.java)
```
Comment (DataM0del) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/DeepL.kt]:
```suggestion
        val inputStream = connection.inputStream
        val reader = inputStream.bufferedReader()
        val gson = GsonBuilder().create()
        val responseObj = gson.fromJson(reader, ResponseData::class.java)
```

**Comment by Razzy52** [comment]
bro its christmas do something else

**Comment by DataM0del** [comment]
> bro its christmas do something else

Something is better than nothing.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/ResponseData.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (DataM0del) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/ResponseData.kt]:
```suggestion
    @get:SerializedName("detected_source_language")
    val detectedSourceLanguage: String
```

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/ResponseData.kt [hunk 4]
```diff
@@ -0,0 +1,9 @@
+package net.ccbluex.liquidbounce.features.module.modules.misc.betterchat.translate.providers.deepl
+
```
Comment (DataM0del) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/misc/betterchat/translate/providers/deepl/ResponseData.kt:5]:
```suggestion
import com.google.gson.annotations.SerializedName

```

**Pull request closed without merging**